api/lib/validate.js

```diff
@@ -166,8 +178,95 @@ function validateOrderbook(orderbook) {
   }
 }
 
+function validateSettings(settings) {
+  if (typeof settings !== 'object') {
+    throw error('Invalid parameter: settings');
```

Comment:
> A javascript client would handle them with try/catch

These are async functions, with a callback that gets passed an error. What would it look like if I were to use something like `async.parallel()` to run several of these in parallel?